### DISCUSSION TITLE
[Question] how validate only numbers or letters in TextField ??

### DISCUSSION THREAD

**Discussion by avilca** [flet-dev/flet#2031] [Q&A] [answered]
### Question

tengo dos textbox codpro  and nompro.
codpro es el código del producto y quisiera que solo permita números estuve revisando la documentación de Flet y vi que  Keyboard_type  podria funcionar pero provee y no funciono.

I have two Textbox CodPro and NomPro. codpro is the code of the product and I would like it to only allow numbers I was reviewing the documentation of Flet and I saw that it could Keyboard_type work but it provides and it did not work. I would like to do the same for nompro that only allows me texts

Lo mismo quisiera hacer para nompro que solo me permita textos
I would like to do the same for nompro that only allows me texts(letters)

Alguna sugerencia como lo podría hacer ??

Any suggestions how I could do it??

### Code sample

```python
def main(page):

    #Tamaño de ventana 
    page.window_resizable = False
        

    #CONTROLES (Controls)

    codpro = TextField(label="CÓDIGO PRODUCTO", width = 200, keyboard_type = KeyboardType.NUMBER)
    nompro = TextField(label="NOMBRE PRODUCTO", width = 600)


   #combo de categoria (Category)
    ddcate = Dropdown(label="CATEGORIA",
        width=150,
        options=[
            dropdown.Option("dulce"),
            dropdown.Option("salado"),
        ],
    )

    #combo de peso (weight)
    ddpeso = Dropdown(label="PESO",
        width=100,
        options=[
            dropdown.Option("KG"),
            ft.dropdown.Option("G"),
        ],
    )

   #Boton AGREGAR
    BtnAgregar = ElevatedButton(
        text="Agregar", 
        bgcolor="green",
        color="white",
        on_click=agregar
        )
    

     page.add(
	    Column([
		    Text("PRODUCTOS",size=30,weight="bold"),
		     codpro, nompro,ddcate, ddpeso,
		     Row([BtnAgregar]),
		     tablaproducto
 
		     ])
 
		)

ft.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
Thats not directly possible at the moment. You can use the `on_change` event to indirectly do this. (although its not that beautiful tbh)
Here is an opened issue for that: https://github.com/flet-dev/flet/issues/631
@FeodorFitsner, did you perhaps see my comment in that issue? (https://github.com/flet-dev/flet/issues/631#issuecomment-1679795943)

**Reply by avilca** [reply]
Thank you @ndonkoHenri
